nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/jdbc/SQLInfo.java

```diff
@@ -1004,12 +1015,28 @@ protected void postProcessSelectChildrenIdsAndTypes() {
                     + getSoftDeleteClause(tableName);
             select.setWhere(where);
             selectChildrenIdsAndTypesSql = select.getStatement();
+            
+            // Build the exclude from copy part of the where clause
+            SchemaManager schemaManager = Framework.getService(SchemaManager.class);
+            List<String> excludedTypes = Stream.of(schemaManager.getExcludedFromCopyDocumentTypes())
+                    .map(d -> d.getName())
+                    .collect(Collectors.toList());
+            String excludeSpecialChildrenClause = " AND " + table.getColumn(Model.MAIN_PRIMARY_TYPE_KEY).getQuotedName() + " NOT IN ('"
+                    + String.join("', '", excludedTypes) + "')";
```

Comment:
Please make sure this doesn't produce invalid SQL if `excludedTypes` is empty (could happen with specific user config or unit tests)